controllers/clusterenvironment_controller.go

```diff
@@ -157,7 +157,8 @@ func (r *ClusterEnvironmentReconciler) Reconcile(ctx context.Context, req ctrl.R
 	errns := r.reconcileNamespaces(ctx, cfg, ce, fann, fsnn)
 	errsns := r.reconcileServiceNamespaces(ctx, cfg, ce, fsnn)
 	errans := r.reconcileApplicationNamespaces(ctx, cfg, ce, fann)
-	if err := errors.Join(errns, errsns, errans); err != nil {
+	errscans := r.reconcileSCatalogApplicationNamespaces(ctx, cfg, ce, fann)
+	if err := errors.Join(errns, errsns, errans, errscans); err != nil {
```

Comment:
I'd propose to modify the function `reconcileApplicationNamespace` such that it calls the two functions `reconcileServiceCatalog` and `reconcileServiceClaims`. Something like the following:

```golang
func (r *ClusterEnvironmentReconciler) reconcileSCatalogApplicationNamespaces(ctx context.Context, cfg *rest.Config, ce *primazaiov1alpha1.ClusterEnvironment, failedApplicationNamespaces []string) error {
        nns := slices.SubtractStr(ce.Spec.ApplicationNamespaces, failedApplicationNamespaces)
        errcm = r.reconcileServiceClaims(ctx, cfg, ce, nns)
        errct = r.reconcileServiceCatalog(ctx, cfg, ce, nns)
        return errors.Join(errcm, errct)
}
```